app/tasks/maintenance_tasks/task.rb

```diff
@@ -71,9 +79,10 @@ def load_constants
     #
     # @raise [NotImplementedError] with a message advising subclasses to
     #   implement an override for this method.
-    def collection
+    def enumerator_builder
       raise NotImplementedError,
-        "#{self.class.name} must implement `collection`."
+        "#{self.class.name} must implement `#{__method__}` or inherit from a class which does"
+      # TODO: Could make error string list available adapters
```

Comment:
We could do something fancy here where we detect a legacy task and provide a useful error message, since if this method is reached from a class implementing `collection`, they're almost certainly using the legacy API.

```ruby
if respond_to?(:collection)
  case collection
  when ActiveRecord::Relation
    raise NotImplementedError, 'Inherit from new ActiveRecordTask instead'
  when Array
    raise NotImplementedError, 'Inherit from new ArrayTask instead'
  else
    raise NotImplementedError, 'Migrate to new Task API for your custom collection'
  end
else
  raise NotImplementedError, "#{self.class.name} must implement `#{__method__}` or inherit from a class which does"
end